### PR TITLE
Feat: persist search history in Redis

### PR DESCRIPTION
**Pull request by timwangmusic**
The `deque` in server has bad performance, resulting in inconsistent read results. Instead of keeping a history in server, persist the results in Redis and keep our previous contract of having a max length of 10 search terms.

### CODE REVIEW

File: iomanagers/redis_manager.py [hunk 1]
```diff
@@ -6,18 +6,28 @@
...
-        self.redis_client.rpush(redis_key, *search_results)
+        self.client.rpush(redis_key, *search_results)
+        self.client.expire(redis_key, self.key_expiration_time)
 
-    # retrieve search results from Redis
     def get_search_results(self, search_term: str) -> list:
         redis_key = "search_term:" + search_term
-        return self.redis_client.lrange(redis_key, 0, -1)
+        return self.client.lrange(redis_key, 0, -1)
+
```
Comment (sananand007) [inline iomanagers/redis_manager.py:25]: [nit] may be start having function docstrings and args have typehints https://docs.python.org/3/library/typing.html

File: iomanagers/redis_manager.py [hunk 2]
```diff
@@ -6,18 +6,28 @@
```
Comment (sananand007) [inline iomanagers/redis_manager.py:7]: [nit] add test for this file and the functions being used

**Review by sananand007 — APPROVED** [review]
lgtm

Overall we should add tests for each file.

File: iomanagers/redis_manager.py (hunk 1 above)
Comment (timwangmusic) [reply iomanagers/redis_manager.py:25]: Will do in another PR. We need to add mocking.

**Pull request merged**